### ISSUE TITLE
Sender component fails to deserialize when opening definition on other computer, clearing the contents and history of the related stream

### ISSUE THREAD

**Issue by PHMRHDHV** [speckleworks/SpeckleAdmin#49]
Step 0: 
- [ ] I've  read the [contribution guidelines](https://github.com/speckleworks/SpeckleOrg/blob/master/CONTRIBUTING.md)!
#### Expected Behaviour
How things should actually work!

#### Actual Behaviour
What's actually going on!

#### Affected Projects
Does this issue propagate to other dependencies or dependants? If so, list them here!

**Comment by PHMRHDHV** [comment]
Hi @didimitrie, the stream is on a private server, unfortunately...
It seems the component failed to find an account and the GET is trying to do /api/accounts/undefined:

error:  {"message":"Cast to ObjectId failed for value \"undefined\" at path \"_id\" for model \"User\"","name":"CastError","stringValue":"\"undefined\"","kind":"ObjectId","value":"undefined","path":"_id"}

The strange thing is that this seems to have emptied the stream and stream history and rendered the stream inaccesible to all other user, except the original stream owner. On the original machine, the stream is still accesible and visible.

**Comment by didimitrie** [comment]
could you send me a screenshot of the accounts popup? though i'm quite sure the problem is not there necessarily. If on the second computer you create a receiver in a separate gh file and try and get that stream, does it work?

**Comment by PHMRHDHV** [comment]
![image](https://user-images.githubusercontent.com/31135559/58029626-44336e00-7b1d-11e9-9c03-35e905f060e7.png)

In principle that works, but the receiver returns a 404 error when linking it to the stream ID.

**Comment by didimitrie** [comment]
this might mean the stream got flagged as deleted? damn this is difficult to debug remotely. does it show up in the admin interface in the "archive" section? On hestia it's https://hestia.speckle.works/#/trash.

**Comment by didimitrie** [comment]
Streams get deleted (or used to get deleted) if you delete the sender component. This might also be getting some sort of confusion when copy pasting grasshopper sender components - though in theory i remember looking at this edge case some time ago and sorting it...

**Comment by PHMRHDHV** [comment]
No, I think I didn't. It doesn't appear in the Trash and is still visible in the original location. It just gave me an empty admin interface, without any source or receivers. Although now, after rerunning the script from the original computer and definition a couple of times, the stream seems to be available again, although with out history and reference to the old data.

**Comment by PHMRHDHV** [comment]
Sorry, forget that last comment, I was logged in to the admin with the account that originally created the stream, there it is properly visible. With the other account it is still inaccesible.

![image](https://user-images.githubusercontent.com/31135559/58030606-20712780-7b1f-11e9-8ba1-5582e070ded4.png)

![image](https://user-images.githubusercontent.com/31135559/58030645-31ba3400-7b1f-11e9-9f70-a18cc3eb76ec.png)

And, although it says that there are three items in the history, nothing is actually there...

**Comment by didimitrie** [comment]
~you might try and refresh the admin ui in "trash" view.~ Gotcha. This is weird. Side note: I also suspect - had a poke at your server url - that the admin ui is out of date, so you might want to update it. 

Last call: if you go to your `server_url/api/streams/your stream id`, can you send me a screenshot of the result? (or could you make the stream public?) 

Another thought: if you create a new stream, with some test data, share it with me and send me that grasshopper definition so i can debug properly? (i've created an account on your server, i should be found if you search for `dimitrie`).

**Comment by PHMRHDHV** [comment]
{"success":false,"message":"Cannot read property 'toString' of null"}

**Comment by didimitrie** [comment]
pfff, that stream somehow got corrupted and would be quite keen to understand why. Theoretically you can restore it from the admin ui; does that change anything?

**Comment by PHMRHDHV** [comment]
I'll give it a try...

**Comment by PHMRHDHV** [comment]
Won't work unfortunately, there is no history or clones anymore...

**Comment by PHMRHDHV** [comment]
Ah, somehow there are no accounts anymore in the pop-up when I insert a new sender component. This might be the source of the issue on the Grasshopper side.

**Comment by didimitrie** [comment]
how did those disappear though!!! there's literally no delete button for them anywhere (besides the rhino client). 

You can check in `%localappdata%/SpeckleSettings/` there should be a sqlite database in there that holds the info on the accounts too (as well as the object caches).

**Comment by PHMRHDHV** [comment]
That seems fine, contents are correct, it has all the accounts...

**Comment by PHMRHDHV** [comment]
I can see it has made three calls with an empty hash.

**Comment by didimitrie** [comment]
is this happening using the plugins that came with the installer, or did you build the plugins yourselves by any chance? 

this is super weird; the accounts should still show up then!

**Comment by PHMRHDHV** [comment]
These are the plugins that came with the installer, i used the url from the server to install them and I'm also using the updater...

**Comment by PHMRHDHV** [comment]
I just re-entered the account and now it appears properly...

**Comment by didimitrie** [comment]
Ok, with your test definition i could partly reproduce this. Main issues I see: 
- core & popup allows you to login to the same server, but with different urls (ie, `/api`, `/api/`,  `/api/v1` - which are equivalent).
- core, in the client deserialisation, looks for an account matching that exact url, and fails if it doesn't, even if the user does have a valid account on that server. 

These, nevertheless, don't amount to the total clusterfuck that happened earlier; ie somehow the stream got flagged as deleted without user input (am i correct in assuming this?). So the quest continues...
